contracts/Commitment/Commitment.sol

```diff
@@ -548,32 +548,26 @@ contract Commitment is
         uint256 amountEur = isEuro ? amount : convertToEur(amount);
         require(amount == 0 || amountEur >= MIN_TICKET_EUR);
 
-        // AUDIT[CHF-42] Protect against reentrancy attack.
-        //   Although the NEUMARK.issueForEuro() is trusted code,
-        //   you should change the order of operations in this function.
-        //   The general rules are described in
-        //   "Order of operations within an external or public function"
-        //   in CodeStyle.md file.
-        //
-        //   At least `_whitelist[investor].token = token` should be set before
-        //   calling NEUMARK.issueForEuro().
-        //
-        //   The proposed code changes are in audit/CHF-42.patch.
-        //
-        //   Also, having a unit test case for reentracy attack on
```

Comment:
While I totally support the pattern to modify state and then call external contracts, in that case re-entrancy is not possible, even if Neumark contract calls back into this or another function, the `msg.sender` will be different and it will be kicked out by `only(ROLE_WHITELIST_ADMIN)` of `addWhitelisted`. Similar situation with other calls to Neumark.